// @sei-js/core/dist/lib/signingClient/stargateClient.js
'use strict';

var _interopRequireDefault = require('@babel/runtime/helpers/interopRequireDefault');
Object.defineProperty(exports, '__esModule', {
	value: true
});
exports.getStargateClient = exports.getSigningClient = exports.createSeiRegistry = exports.createSeiAminoTypes = void 0;
var _regenerator = _interopRequireDefault(require('@babel/runtime/regenerator'));
var _asyncToGenerator2 = _interopRequireDefault(require('@babel/runtime/helpers/asyncToGenerator'));
var _defineProperty2 = _interopRequireDefault(require('@babel/runtime/helpers/defineProperty'));
var _toConsumableArray2 = _interopRequireDefault(require('@babel/runtime/helpers/toConsumableArray'));
var _stargate = require('@cosmjs/stargate');
var _protoSigning = require('@cosmjs/proto-signing');
var _proto = require('@sei-js/proto');
function ownKeys(object, enumerableOnly) {
	var keys = Object.keys(object);
	if (Object.getOwnPropertySymbols) {
		var symbols = Object.getOwnPropertySymbols(object);
		enumerableOnly &&
			(symbols = symbols.filter(function (sym) {
				return Object.getOwnPropertyDescriptor(object, sym).enumerable;
			})),
			keys.push.apply(keys, symbols);
	}
	return keys;
}
function _objectSpread(target) {
	for (var i = 1; i < arguments.length; i++) {
		var source = null != arguments[i] ? arguments[i] : {};
		i % 2
			? ownKeys(Object(source), !0).forEach(function (key) {
					(0, _defineProperty2['default'])(target, key, source[key]);
			  })
			: Object.getOwnPropertyDescriptors
			? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source))
			: ownKeys(Object(source)).forEach(function (key) {
					Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
			  });
	}
	return target;
}
var createSeiRegistry = function createSeiRegistry() {
	return new _protoSigning.Registry(
		[].concat(
			(0, _toConsumableArray2['default'])(_stargate.defaultRegistryTypes),
			(0, _toConsumableArray2['default'])(_proto.cosmwasmProtoRegistry),
			(0, _toConsumableArray2['default'])(_proto.seiprotocolProtoRegistry)
		)
	);
};
exports.createSeiRegistry = createSeiRegistry;
var createSeiAminoTypes = function createSeiAminoTypes() {
	var types = _objectSpread(
		_objectSpread(_objectSpread(_objectSpread({}, _proto.cosmosAminoConverters), _proto.cosmwasmAminoConverters), _proto.ibcAminoConverters),
		_proto.seiprotocolAminoConverters
	);
	return new _stargate.AminoTypes(types);
};
exports.createSeiAminoTypes = createSeiAminoTypes;
var getStargateClient = /*#__PURE__*/ (function () {
	var _ref = (0, _asyncToGenerator2['default'])(
		/*#__PURE__*/ _regenerator['default'].mark(function _callee(rpcEndpoint) {
			var options,
				_args = arguments;
			return _regenerator['default'].wrap(function _callee$(_context) {
				while (1)
					switch ((_context.prev = _context.next)) {
						case 0:
							options = _args.length > 1 && _args[1] !== undefined ? _args[1] : {};
							return _context.abrupt('return', _stargate.StargateClient.connect(rpcEndpoint, options));
						case 2:
						case 'end':
							return _context.stop();
					}
			}, _callee);
		})
	);
	return function getStargateClient(_x) {
		return _ref.apply(this, arguments);
	};
})();
exports.getStargateClient = getStargateClient;
var getSigningClient = /*#__PURE__*/ (function () {
	var _ref2 = (0, _asyncToGenerator2['default'])(
		/*#__PURE__*/ _regenerator['default'].mark(function _callee2(rpcEndpoint, signer) {
			var options,
				registry,
				aminoTypes,
				_args2 = arguments;
			return _regenerator['default'].wrap(function _callee2$(_context2) {
				while (1)
					switch ((_context2.prev = _context2.next)) {
						case 0:
							options = _args2.length > 2 && _args2[2] !== undefined ? _args2[2] : {};
							registry = createSeiRegistry();
							aminoTypes = createSeiAminoTypes();
							return _context2.abrupt(
								'return',
								_stargate.SigningStargateClient.connectWithSigner(
									rpcEndpoint,
									signer,
									_objectSpread(
										{
											registry: registry,
											aminoTypes: aminoTypes
										},
										options
									)
								)
							);
						case 4:
						case 'end':
							return _context2.stop();
					}
			}, _callee2);
		})
	);
	return function getSigningClient(_x2, _x3) {
		return _ref2.apply(this, arguments);
	};
})();
exports.getSigningClient = getSigningClient;
